src/toolkits/style_transfer/style_transfer.cpp

```diff
@@ -0,0 +1,354 @@
+  const size_t batch_size = batch.size();
+
+  std::vector<float> content_array(height * width * channels * batch.size());
+  std::vector<float> style_array(height * width * channels * batch.size());
+  std::vector<float> index_array(batch_size);
+
+  for (size_t index = 0; index < batch_size; index++) {
+    prepare_images(batch[index].content_image, content_array, width, height,
+                   channels, index);
+
+    prepare_images(batch[index].style_image, style_array, width, height,
+                   channels, index);
+
+    size_t style_index = batch[index].style_index;
+    index_array[index] = style_index;
+  }
+
+  return {
+    {"input", shared_float_array::wrap(std::move(content_array),
+                                       {batch_size, height, width, channels})},
+    {"labels", shared_float_array::wrap(std::move(style_array),
+                                       {batch_size, height, width, channels})},
+    {"index", shared_float_array::wrap(std::move(index_array),
+                                       {batch_size})}
+  };
+}
+
+flex_int estimate_max_iterations(flex_int num_styles, flex_int batch_size) {
+  return static_cast<flex_int>(num_styles * 10000.0f / batch_size);
+}
+
+}  // namespace
+
+void style_transfer::init_options(
+    const std::map<std::string, flexible_type>& opts) {
+  options.create_integer_option(
+      "batch_size",
+      "The number of images to process for each training iteration",
+      FLEX_UNDEFINED, 1, std::numeric_limits<int>::max());
+
+  options.create_integer_option(
+      "max_iterations",
+      "Maximum number of iterations to perform during training", FLEX_UNDEFINED,
+      1, std::numeric_limits<int>::max());
+
+  options.create_integer_option(
+      "image_width", "The width of the images passed into the network",
+      FLEX_UNDEFINED, 1, std::numeric_limits<int>::max());
+
+  options.create_integer_option(
+      "image_height", "The height of the images passed into the network",
+      FLEX_UNDEFINED, 1, std::numeric_limits<int>::max());
+
+  options.create_integer_option(
+      "random_seed",
+      "Seed for random weight initialization and sampling during training",
+      FLEX_UNDEFINED, std::numeric_limits<int>::min(),
+      std::numeric_limits<int>::max());
+
+  options.create_integer_option(
+      "num_styles", "The number of styles present in the model", FLEX_UNDEFINED,
+      1, std::numeric_limits<int>::max());
+
+  options.set_options(opts);
+
+  add_or_update_state(flexmap_to_varmap(options.current_option_values()));
+}
+
+size_t style_transfer::get_version() const { return STYLE_TRANSFER_VERSION; }
+
+void style_transfer::save_impl(oarchive& oarc) const {
+  variant_deep_save(state, oarc);
+  oarc << m_resnet_spec->export_params_view()
+       << m_vgg_spec->export_params_view();
+}
+
+void style_transfer::load_version(iarchive& iarc, size_t version) {
+  variant_deep_load(state, iarc);
+
+  float_array_map nn_params;
+  iarc >> nn_params;
```

Comment:
You wrote two `float_array_map` values in `save_impl` but only read one `float_array_map` value here.... Did you mean to still write the VGG spec?

Reply:
I removed it, must have slipped my mind thanks!